app/src/main/java/com/example/taskwise_eventmaster/data/persistance/daos/EventDao.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.taskwise_eventmaster.data.persistance.daos
+
+import androidx.room.Dao
+import androidx.room.Query
+import androidx.room.Upsert
+import com.example.taskwise_eventmaster.data.persistance.model.Event
+
+@Dao
+interface EventDao {
+    @Upsert
+    suspend fun upsertEvent(event: Event)
+
+    @Query("DELETE FROM Event WHERE id=:eventId")
+    suspend fun deleteEvent(eventId: Int)
+
+    @Query("SELECT * FROM Event WHERE id=:eventId")
+    fun getEventById(eventId: Int): Event?
+
+    @Query("SELECT * FROM Event")
+    fun getAllEventsLocal(): List<Event>
```

Comment:
No need to mention `Local` in the naming here as it is implied by fetching from the local db